libnetwork/sandbox_store.go

```diff
@@ -28,9 +28,16 @@ type sbState struct {
 	EpPriority map[string]int
 	// external servers have to be persisted so that on restart of a live-restore
 	// enabled daemon we get the external servers for the running containers.
-	// We have two versions of ExtDNS to support upgrade & downgrade of the daemon
-	// between >=1.14 and <1.14 versions.
-	ExtDNS  []string
+	ExtDNS []extDNSEntry
+	// ExtDNS2 was used to handle migration between docker < 1.14 and >= 1.14.
+	// Before version 1.14 we used ExtDNS but with a []string. As it's unlikely
+	// that installations still have state from before 1.14, we're renaming
+	// ExtDNS2 back to ExtDNS. Keeping both ExtDNS and ExtDNS2 to allow migrating
+	// state from older version.
+	//
+	// Deprecated: only present to migrate existing state. This field will be removed in a future version.
+	//
+	// TODO(thaJeztah): remove this field once we can assume no users have state left from old versions.
 	ExtDNS2 []extDNSEntry
```

Comment:
Why bother with new migration code at all? The persisted state is merely an implementation detail so I don't see enough benefit to warrant the churn and complexity.

```suggestion
	ExtDNS []extDNSEntry `json:"ExtDNS2"`
```

Reply:
Would it be confusing if someone would be looking at the JSON and the fieldnames don't match?

Reply:
Okay, let's take the easy approach (we also discussed Mirantis skipping releases, which would mean we need to keep the migration code for a long time)